### PR TITLE
Create topics when bootstrapping Tamer

### PR DESCRIPTION
**Pull request by sirocchj**
And some more config

### CODE REVIEW

File: db/local/docker-compose.yml [hunk 1]
```diff
@@ -43,47 +43,16 @@ services:
     networks:
       - local_kafka_network
     environment:
+      - KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1
+      - KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR=1
+      - KAFKA_TRANSACTION_STATE_LOG_MIN_ISR=1
```
Comment (sirocchj) [inline db/local/docker-compose.yml:48]: these are needed to use confluent's image w/ one kafka only

File: db/local/runDatabaseSimple.sh [hunk 2]
```diff
@@ -13,10 +13,16 @@ export KAFKA_SCHEMA_REGISTRY_URL=http://localhost:8081
 export KAFKA_CLOSE_TIMEOUT=10s
 export KAFKA_BUFFER_SIZE=50
 export KAFKA_SINK_TOPIC=sink
+export KAFKA_SINK_AUTO_CREATE=on
+export KAFKA_SINK_PARTITIONS=1
+export KAFKA_SINK_REPLICAS=1
```
Comment (sirocchj) [inline db/local/runDatabaseSimple.sh:18]: these 3 things control whether a topic is created or not

File: db/local/runDatabaseSimple.sh [hunk 3]
```diff
@@ -13,10 +13,16 @@ export KAFKA_SCHEMA_REGISTRY_URL=http://localhost:8081
...
 export KAFKA_STATE_TOPIC=state
-export KAFKA_STATE_GROUP_ID=state-group
-export KAFKA_STATE_CLIENT_ID=state-client
```
Comment (sirocchj) [inline db/local/runDatabaseSimple.sh:18]: note how these 2 have changed name, this is a breaking change

File: core/src/main/scala/tamer/config.scala [hunk 4]
```diff
@@ -53,6 +69,8 @@ object KafkaConfig {
     bufferSize = bufferSize,
     sink = sink,
     state = state,
+    groupId = groupId,
+    clientId = clientId,
```
Comment (sirocchj) [inline core/src/main/scala/tamer/config.scala:73]: these aren't inherently properties of the state, but more of a consumer, lifting here. breaking change as it impacts the way the config is looked for (hence the env variable)

File: core/src/main/scala/tamer/config.scala [hunk 5]
```diff
@@ -1,39 +1,53 @@
...
 import zio._
 
-final case class RegistryConfig(url: String, cacheSize: Int)
+final case class RegistryConfig(url: String, cacheSize: Int, expiration: JDuration)
 object RegistryConfig {
   def apply(url: String): RegistryConfig = RegistryConfig(
     url = url,
-    cacheSize = 1000
+    cacheSize = 4,
+    expiration = 1.hour
```
Comment (sirocchj) [inline core/src/main/scala/tamer/config.scala:12]: more sensible cache size for our use case and an expiration that is configurable

**Review by amir — APPROVED** [review]

**Pull request merged**